### PR TITLE
Add array, boolean, and integer koans

### PR DESCRIPTION
**Pull request by corincerami**

### CODE REVIEW

**Comment by corincerami** [comment]
@crazymykl Your additions have been merged in so this should all work at the point.

**Comment by crazymykl** [comment]
Maybe make each function in each file its own test? Also, `path_to_enlightenment.rs` is generated, we should remove it from the repo and `.gitignore` it. Everything else looks good, although after you make my changes, maybe rebase it into fewer commits.

**Comment by corincerami** [comment]
@crazymykl History has been rewritten! What is the world coming to?

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 target
+Cargo.lock
```
Comment (crazymykl) [inline .gitignore]: We are building an application, not a library. Much as with Rails, we want the lock checked in in this circumstance.

**Comment by corincerami** [comment]
@crazymykl Now the world can see our locks.

**Pull request merged**